plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -747,11 +725,17 @@ void state_history_plugin::plugin_shutdown() {
    my->applied_transaction_connection.reset();
    my->accepted_block_connection.reset();
    my->block_start_connection.reset();
-   while (!my->sessions.empty())
-      my->sessions.begin()->second->close();
+   for (auto [ session, _]: my->sessions) {
+      session->close();
```

Comment:
This callback to session erase is rather fragile. Please cleanup as we discussed.